src/generic/srchctlg.cpp

```diff
@@ -222,6 +222,14 @@ class wxSearchButton : public wxControl
     void OnPaint(wxPaintEvent&)
     {
         wxPaintDC dc(this);
+
+        // Clear the background in case of a user bitmap with alpha channel
+        wxColour bg = m_search->GetBackgroundColour();
+        dc.SetBrush(wxBrush(bg));
+        dc.SetPen(wxPen(bg));
```

Comment:
It is unnecessary for the **built-in** **Search** and **Cancel** bitmaps because it is done in the `wxBitmap wxSearchCtrl::RenderSearchBitmap( int x, int y, bool renderDrop )`.
**BUT** if you try to use your own bitmaps using `void wxSearchCtrl::SetSearchBitmap( const wxBitmap& bitmap )` or `void wxSearchCtrl::SetCancelBitmap( const wxBitmap& bitmap )`, and these bitmaps would have alpha channel, then in this case the background will be messy.